src/did/did-resolver.ts

```diff
@@ -36,7 +38,14 @@ export class DidResolver {
       throw new Error(`${didMethod} DID method not supported`);
     }
 
-    const resolutionResult = await didResolver.resolve(did);
+    const resolutionResult = await this.memoryCache.has(did) ? await this.memoryCache.get(did): await didResolver.resolve(did);
```

Comment:
There is no need to introduce `.has()`, `.get()` will return `undefined` if not in cache, you can use that as your condition.

Reply:
i did use that first but thought the .has would be more clean

Reply:
Probably logically more clean, but "dirtier" in practice as it requires two round trips to the cache.